cpp/models/abm/testing_strategy.cpp

```diff
@@ -105,12 +102,20 @@ void TestingScheme::update_activity_status(TimePoint t)
 
 bool TestingScheme::run_scheme(PersonalRandomNumberGenerator& rng, Person& person, TimePoint t) const
 {
-    if (t - person.get_time_of_last_test() > m_minimal_time_since_last_test) {
-        if (m_testing_criteria.evaluate(person, t)) {
-            double random = UniformDistribution<double>::get_instance()(rng);
-            if (random < m_probability) {
-                return !person.get_tested(rng, t, m_test_parameters);
-            }
+    auto test_result = person.get_test_result(m_test_parameters.type);
```

Comment:
What happens here if there has not been a test and the entry is empty?

Reply:
If the m_test_parameters is initialised correctly, all the `TestType `should have a result.